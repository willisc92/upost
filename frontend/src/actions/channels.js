import API from "../utils/API";

// SET_CHANNELS
export const setChannels = (channels) => ({
    type: "SET_CHANNELS",
    channels
});

// START_SET_CHANNELS
export const startSetChannels = () => {
    return (dispatch) => {
        API.get("channels/", {
            params: {
                user: localStorage.getItem("user_id")
            }
        }).then((result) => {
            dispatch(setChannels(result.data));
        });
    };
};

export const addChannel = (channel) => {
    return (dispatch) => {
        API.post("channels/", {
            ...channel,
            user: localStorage.getItem("user_id")
        })
            .then((result) => {
                dispatch(startSetChannels());
            })
            .catch((err) => {
                alert(err);
            });
    };
};
