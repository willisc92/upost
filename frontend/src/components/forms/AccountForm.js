import React from "react";
import { connect } from "react-redux";
import DatePicker from "react-date-picker";
import moment from "moment";
import API from "../../utils/API";

export class AccountForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            maxDate: new Date(),
            minDate: new Date(1900, 1, 1),
            birthDate: new Date(),
            isLoaded: false,
            interests: [],
            error: undefined,
            selectedInterests: []
        };
    }

    onDateChange = (date) => {
        this.setState({
            birthDate: date
        });
    };

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };

    componentDidMount() {
        API.get("interests/").then(
            (result) => {
                const interests = result.data.map((interest) => {
                    return { value: interest.interest_tag, label: interest.interest_tag };
                });
                this.setState({
                    isLoaded: true,
                    interests: interests
                });
            },

            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                });
            }
        );
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({
            formInput: e.target,
            interests: this.state.selectedInterests.map((interest) => {
                return interest.value;
            }),
            birthDate: this.state.birthDate
        });
    };

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                <p>Username*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.username &&
                    this.props.auth.error.username.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="text" className="text-input" name="username" />
                <p>150 characters of fewer. Letters, digits and @/./+/-/) only.</p>
                <p>Email Address</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.email &&
                    this.props.auth.error.email.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="text" className="text-input" name="email" />
                <p>First Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.first_name &&
                    this.props.auth.error.first_name.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="text" className="text-input" name="firstName" />
                <p>Middle Name</p>
                <input type="text" className="text-input" name="middleName" />
                <p>Last Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.last_name &&
                    this.props.auth.error.last_name.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="text" className="text-input" name="lastName" />
                <p>Birth date</p>
                <DatePicker
                    value={this.state.birthDate}
                    onChange={this.onDateChange}
                    maxDate={this.state.maxDate}
                    minDate={this.state.minDate}
                    minDetail="decade"
                />
                <p>Password</p>
                <p>Your password contain at least {this.props.minPasswordLength} characters.</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="password" className="text-input" name="password" />
                <p>Password Confirmation*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return (
                            <p className="form__error" key={error}>
                                {error}
                            </p>
                        );
                    })}
                <input type="password" className="text-input" name="passwordConfirmation" />
                <br />
                <button className="button">Sign Up</button>
            </form>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        auth: state.auth
    };
};

export default connect(mapStateToProps)(AccountForm);
