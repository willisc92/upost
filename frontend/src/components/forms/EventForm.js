import React from "react";
import { connect } from "react-redux";
import DateTimePicker from "react-datetime-picker";

export class EventForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            post: !!this.props.event ? this.props.event.post : this.props.post,
            location: !!this.props.event ? this.props.event.location : "",
            capacity: !!this.props.event ? this.props.event.capacity : "0",
            error: "",
            startDate: !!this.props.event ? new Date(this.props.event.planned_start_date) : new Date(),
            endDate: !!this.props.event
                ? new Date(this.props.event.planned_end_date)
                : new Date(new Date().setHours(new Date().getHours() + 1))
        };
    }

    determineReadOnly = () => {
        if (this.props.event) {
            if (new Date(this.props.event.startDate) >= new Date()) {
                return true;
            } else {
                return false;
            }
        } else {
            return false;
        }
    };

    onLocationChange = (e) => {
        const location = e.target.value;
        if (!!location) {
            if (location.length > 50) {
                this.setState({ error: "Must have a location length of 50 characters or less" });
            } else {
                this.setState(() => ({ location }));
            }
        } else {
            this.setState(() => ({ location }));
        }
    };

    onCapacityChange = (e) => {
        const capacity = e.target.value;
        if (!!capacity) {
            this.setState(() => ({ capacity }));
        }
    };

    onStartDateChange = (startDate) => {
        this.setState(() => ({
            startDate
        }));
    };

    onEndDateChange = (endDate) => {
        this.setState(() => ({
            endDate
        }));
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.location) {
            this.setState(() => ({ error: "Please enter a location/room for your event." }));
        } else if (!this.state.capacity) {
            this.setState(() => ({ error: "Please enter a capacity for your event." }));
        } else if (this.state.endDate <= this.state.startDate) {
            this.setState(() => ({ error: "End datetime must be after start datetime." }));
        } else {
            this.setState(() => ({ error: "" }));
            this.props.onSubmit({
                post: this.state.post,
                user: localStorage.getItem("user_id"),
                location: this.state.location,
                capacity: this.state.capacity,
                planned_start_date: this.state.startDate,
                planned_end_date: this.state.endDate
            });
        }
    };

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit} id={this.props.id}>
                {!!this.props.eventError && <p className="form__error">Request failed...</p>}
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <p className="form__error">* - Fields required</p>
                <div className="input-group">
                    <p className="form__label">Location/Room *:</p>
                    <input
                        readOnly={this.determineReadOnly()}
                        className="text-input"
                        type="text"
                        placeholder="Location"
                        autoFocus
                        value={this.state.location}
                        onChange={this.onLocationChange}
                    />
                </div>
                <div className="input-group">
                    <p className="form__label">Capacity (Select 0 if undefined) *: </p>
                    <input
                        readOnly={this.determineReadOnly()}
                        className="text-input"
                        type="number"
                        value={this.state.capacity}
                        onChange={this.onCapacityChange}
                        min="0"
                    />
                </div>
                <div className="input-group">
                    <p className="form__label">Start Date *:</p>
                    <DateTimePicker
                        disabled={this.determineReadOnly()}
                        onChange={this.onStartDateChange}
                        value={this.state.startDate}
                        clearIcon={null}
                        minDate={new Date()}
                    />
                    <div />
                </div>
                <div className="input-group">
                    <p className="form__label">End Date *:</p>
                    <DateTimePicker
                        disabled={this.determineReadOnly()}
                        onChange={this.onEndDateChange}
                        value={this.state.endDate}
                        clearIcon={null}
                        minDate={this.state.startDate}
                    />
                    <div />
                </div>
                <div>
                    <button className="button">{this.props.nextStep}</button>
                </div>
            </form>
        );
    }
}

const mapStateToProps = (state) => ({
    eventError: !!state.events.error && state.events.error.response.data
});

export default connect(mapStateToProps)(EventForm);
