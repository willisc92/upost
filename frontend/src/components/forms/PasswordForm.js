import React from "react";

class PasswordForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            password: "",
            confirm: ""
        };
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onConfirmChange = (e) => {
        const confirm = e.target.value;
        this.setState(() => ({ confirm }));
    };

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({ password: this.state.password, confirm: this.state.confirm });
    };

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit} id={this.props.id}>
                {!!this.props.error && <p className="form__error">{this.props.error}</p>}
                <input
                    type="password"
                    className="text-input"
                    name="password"
                    placeholder={`Password* (At least ${this.props.minPasswordLength} characters)`}
                    value={this.state.password}
                    onChange={this.onPasswordChange}
                />
                <input
                    type="password"
                    className="text-input"
                    name="passwordConfirmation"
                    placeholder="Password Confirmation*"
                    value={this.state.confirm}
                    onChange={this.onConfirmChange}
                />
            </form>
        );
    }
}

export default PasswordForm;
