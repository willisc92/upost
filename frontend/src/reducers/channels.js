// Channels Reducers
const channelsDefaultState = [];

export default (state = channelsDefaultState, action) => {
    switch (action.type) {
        case "SET_CHANNELS":
            return {
                channels: action.channels
            };
        default:
            return state;
    }
};
