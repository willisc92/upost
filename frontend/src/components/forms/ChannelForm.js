import React from "react";
import { connect } from "react-redux";
import { getCurrentUser } from "../../actions/auth";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";

export class ChannelForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            channel_name: this.props.channel ? this.props.channel.channel_name : "",
            channel_description: this.props.channel ? this.props.channel.channel_description : "",
            error: ""
        };
    }

    onNameChange = (e) => {
        const channel_name = e.target.value;
        if (!!channel_name) {
            if (channel_name.length > 50) {
                this.setState({ error: "Must have a channel name 50 characters or less" });
            } else {
                this.setState(() => ({ channel_name }));
            }
        } else {
            this.setState(() => ({ channel_name }));
        }
    };

    onDescriptionChange = (e) => {
        const channel_description = e.target.value;
        if (!!channel_description) {
            if (channel_description.length > 500) {
                this.setState({ error: "Must have a channel description 500 characters or less" });
            } else {
                this.setState(() => ({ channel_description }));
            }
        } else {
            this.setState(() => ({ channel_description }));
        }
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.channel_name) {
            this.setState(() => ({ error: "Please provide a channel name" }));
        } else {
            getCurrentUser().then((res) => {
                this.setState(() => ({ error: "" }));
                this.props.onSubmit({
                    channel_name: this.state.channel_name,
                    channel_description: this.state.channel_description,
                    user: res.data.username
                });
            });
        }
    };

    render() {
        return (
            <div>
                <form className="form" onSubmit={this.onSubmit}>
                    {!!this.props.error && !!this.props.error.channel_name && (
                        <Box paddingBottom={3}>
                            <Typography color="error" variant="h2">
                                {this.props.error.channel_name[0]}
                            </Typography>
                        </Box>
                    )}
                    {this.state.error && (
                        <Box paddingBottom={3}>
                            <Typography color="error" variant="h2">
                                {this.state.error}
                            </Typography>
                        </Box>
                    )}

                    <Box py={2}>
                        <TextField
                            label="Name"
                            type="text"
                            placeholder="Name"
                            value={this.state.channel_name}
                            onChange={this.onNameChange}
                            disabled={this.props.read_only}
                            required
                        />
                    </Box>
                    <Box py={2}>
                        <TextField
                            label="Description"
                            className="text-input"
                            type="text"
                            placeholder="Description"
                            value={this.state.channel_description}
                            onChange={this.onDescriptionChange}
                            disabled={this.props.read_only}
                            multiline
                        />
                    </Box>
                    {!this.props.read_only && (
                        <Box>
                            <Button color="primary" variant="contained" type="submit">
                                Submit
                            </Button>
                        </Box>
                    )}
                </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    error: !!state.channels.error && state.channels.error.response.data
});

export default connect(mapStateToProps)(ChannelForm);
