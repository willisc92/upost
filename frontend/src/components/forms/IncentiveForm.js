import React from "react";
import { connect } from "react-redux";
import { startGetIncentiveTypes } from "../../actions/incentive_types";
import { startGetDietOptions } from "../../actions/diet_options";
import { getCurrentUser } from "../../actions/auth";
import moment from "moment";

import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import { DateTimePicker } from "@material-ui/pickers";
import MenuItem from "@material-ui/core/MenuItem";
import Checkbox from "@material-ui/core/Checkbox";
import ListItemText from "@material-ui/core/ListItemText";
import { MultiSelect } from "../Select";

class IncentiveForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            fromPost: !!this.props.fromPost,
            fromEvent: !!this.props.fromEvent,
            post: !!this.props.incentivePackage
                ? this.props.incentivePackage.post
                : !!this.props.post
                ? this.props.post
                : null,
            event: !!this.props.incentivePackage
                ? this.props.incentivePackage.event
                : !!this.props.event
                ? this.props.event.event_id
                : null,
            diet_option: !!this.props.incentivePackage ? this.props.incentivePackage.diet_option : [],
            incentive_type: !!this.props.incentivePackage ? this.props.incentivePackage.incentive_type : [],
            ip_description: !!this.props.incentivePackage
                ? this.props.incentivePackage.ip_description
                : !!this.props.description
                ? this.props.description
                : "",
            planned_start_date:
                !!this.props.incentivePackage && !!this.props.incentivePackage.planned_start_date
                    ? moment(this.props.incentivePackage.planned_start_date)
                    : !!this.props.event && !!this.props.event.planned_start_date
                    ? moment(this.props.event.planned_start_date)
                    : !!this.props.fromEvent
                    ? moment()
                    : null,
            planned_end_date:
                !!this.props.incentivePackage && !!this.props.incentivePackage.planned_end_date
                    ? moment(this.props.incentivePackage.planned_end_date)
                    : !!this.props.event && !!this.props.event.planned_end_date
                    ? moment(this.props.event.planned_end_date)
                    : !!this.props.fromEvent
                    ? moment().add(1, "hours")
                    : null
        };
    }

    componentWillMount() {
        this.props
            .startGetIncentiveTypes()
            .then(() => {})
            .catch((error) => {
                console.log(JSON.stringify(error, null, 2));
            });
        this.props
            .startGetDietOptions()
            .then(() => {})
            .catch((error) => {
                console.log(JSON.stringify(error, null, 2));
            });
    }

    onDescriptionChange = (e) => {
        const ip_description = e.target.value;
        if (!!ip_description) {
            if (ip_description.length > 500) {
                this.setState({ error: "Description must be 500 characters or less" });
            } else {
                this.setState(() => ({ ip_description }));
            }
        } else {
            this.setState(() => ({ ip_description }));
        }
    };

    onDietOptionsChange = (e) => {
        this.setState(() => ({ diet_option: e.target.value }));
    };

    onIncentiveTypeChange = (e) => {
        e.persist();

        this.setState(() => ({
            incentive_type: e.target.value
        }));
    };

    onStartDateChange = (planned_start_date) => {
        this.setState(() => ({
            planned_start_date
        }));
    };

    onEndDateChange = (planned_end_date) => {
        this.setState(() => ({
            planned_end_date
        }));
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (this.state.incentive_type.length === 0) {
            this.setState(() => ({ error: "Please select at least one incentive type." }));
        } else if (!this.state.ip_description) {
            this.setState(() => ({ error: "Please enter a brief description." }));
        } else if (
            !!this.state.planned_end_date &&
            !!this.state.planned_start_date &&
            this.state.planned_end_date <= this.state.planned_start_date
        ) {
            this.setState(() => ({ error: "End datetime must be after start datetime." }));
        } else if (this.state.incentive_type.includes("Food") && this.state.diet_option.length === 0) {
            this.setState(() => ({ error: "Must have at least one diet option selected" }));
        } else {
            getCurrentUser()
                .then((res) => {
                    this.setState(() => ({ error: "" }));
                    const payload = {
                        post: this.state.post,
                        event: this.state.event,
                        user: res.data.id,
                        diet_option: this.state.diet_option,
                        incentive_type: this.state.incentive_type,
                        ip_description: this.state.ip_description,
                        planned_start_date: !!this.state.planned_start_date
                            ? this.state.planned_start_date.toDate()
                            : null,
                        planned_end_date: !!this.state.planned_end_date ? this.state.planned_end_date.toDate() : null
                    };
                    this.props.onSubmit(payload);
                    console.log(payload);
                })
                .catch((err) => {
                    console.log(err);
                    console.log(JSON.stringify(err, null, 2));
                });
        }
    };

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit} id={this.props.id}>
                <Box py={2}>
                    {!!this.props.incentivePackageError && (
                        <Typography variant="h2" color="error" gutterBottom>
                            Request failed...
                        </Typography>
                    )}
                    {!!this.state.error && (
                        <Typography variant="h2" color="error" gutterBottom>
                            {this.state.error}
                        </Typography>
                    )}
                </Box>
                {!!this.props.incentiveTypes && (
                    <Box display="flex">
                        <Box paddingRight={2}>
                            <Typography>Perk Types:</Typography>
                        </Box>
                        <MultiSelect
                            required
                            disabled={this.props.read_only}
                            onChange={this.onIncentiveTypeChange}
                            value={this.state.incentive_type}
                            label="Perks"
                        >
                            {this.props.incentiveTypes.map((incentiveType) => {
                                return (
                                    <MenuItem key={incentiveType.incentive_name} value={incentiveType.incentive_name}>
                                        <Checkbox
                                            color="primary"
                                            checked={
                                                this.state.incentive_type.indexOf(incentiveType.incentive_name) !== -1
                                            }
                                        />
                                        <ListItemText primary={incentiveType.incentive_name} />
                                    </MenuItem>
                                );
                            })}
                        </MultiSelect>
                    </Box>
                )}
                {this.state.incentive_type.includes("Food") && (
                    <Box display="flex">
                        <Box paddingRight={2}>
                            <Typography>Diet Options*:</Typography>
                        </Box>
                        <MultiSelect
                            required
                            label="Diet Options"
                            onChange={this.onDietOptionsChange}
                            value={this.state.diet_option}
                            disabled={this.props.read_only}
                        >
                            {this.props.dietOptions.map((diet_option) => {
                                return (
                                    <MenuItem key={diet_option.diet_option} value={diet_option.diet_option}>
                                        <Checkbox
                                            color="primary"
                                            checked={this.state.diet_option.indexOf(diet_option.diet_option) !== -1}
                                        />
                                        <ListItemText primary={diet_option.diet_option} />
                                    </MenuItem>
                                );
                            })}
                        </MultiSelect>
                    </Box>
                )}
                <Box>
                    <TextField
                        required
                        autoFocus
                        label="Perk Package Description"
                        placeholder={`Describe the perks for your ${!!this.state.fromPost ? "post" : "event"}.`}
                        multiline
                        value={this.state.ip_description}
                        onChange={this.onDescriptionChange}
                        disabled={this.props.read_only}
                    />
                </Box>
                {!!this.state.fromEvent && (
                    <React.Fragment>
                        <Box>
                            <DateTimePicker
                                value={this.state.planned_start_date}
                                onChange={this.onStartDateChange}
                                label="Start Date"
                                disabled={this.props.read_only}
                                required
                            />
                        </Box>
                        <Box>
                            <DateTimePicker
                                value={this.state.planned_end_date}
                                onChange={this.onEndDateChange}
                                label="End Date"
                                disabled={this.props.read_only}
                                required
                            />
                        </Box>
                    </React.Fragment>
                )}
                <div>
                    {!this.props.read_only && (
                        <Button color="primary" variant="contained" type="submit">
                            {this.props.nextStep}
                        </Button>
                    )}
                </div>
            </form>
        );
    }
}

const mapStateToProps = (state) => ({
    loading:
        !!state.incentivetypes && !!state.dietOptions && (state.incentiveTypes.loading || state.dietOptions.loading),
    incentiveTypesError: !!state.incentiveTypes.error && state.incentiveTypes.error.response.data,
    incentiveTypes: !!state.incentiveTypes && state.incentiveTypes.incentiveTypes,
    dietOptionsError: !!state.dietOptions.error && state.dietOptions.error.response.data,
    dietOptions: !!state.dietOptions && state.dietOptions.dietOptions,
    incentivePackageError: !!state.incentivePackage.error && state.incentivePackage.error.response.data
});

const mapDispatchToProps = (dispatch) => ({
    startGetIncentiveTypes: () => dispatch(startGetIncentiveTypes()),
    startGetDietOptions: () => dispatch(startGetDietOptions())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(IncentiveForm);
