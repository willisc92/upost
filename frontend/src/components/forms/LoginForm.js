import React from "react";
import { connect } from "react-redux";
import { authLogin } from "../../actions/auth";

export class LoginForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            username: "",
            password: ""
        };
    }

    onUserNameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({ username }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({ username: this.state.username, password: this.state.password });
    };

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {!!this.props.error && <p className="form__error">Request failed.</p>}
                <input
                    className="text-input"
                    type="text"
                    placeholder="Username"
                    autoFocus
                    value={this.state.username}
                    onChange={this.onUserNameChange}
                />
                <input
                    className="text-input"
                    type="password"
                    value={this.state.password}
                    onChange={this.onPasswordChange}
                    placeholder="Password"
                />
                <div className="modal__button">
                    <button className="button">Login</button>
                </div>
            </form>
        );
    }
}

const mapStateToProps = (state) => ({
    error: state.auth.error
});

const mapDispatchToProps = (dispatch) => ({
    authLogin: (username, password) => dispatch(authLogin(username, password))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginForm);
