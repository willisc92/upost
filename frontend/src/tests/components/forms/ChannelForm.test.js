import { ChannelForm } from "../../../components/forms/ChannelForm";
import { shallow } from "enzyme";
import React from "react";
import channels from "../../fixtures/channels";

let wrapper, onSubmitSpy, error;

beforeEach(() => {
    onSubmitSpy = jest.fn();
    error = undefined;
    wrapper = shallow(<ChannelForm error={error} onSubmit={onSubmitSpy} />);
});

test("Should render ChannelForm correctly", () => {
    expect(wrapper).toMatchSnapshot();
});

test("Should render ChannelForm with existing channel data", () => {
    wrapper = shallow(<ChannelForm channel={channels[0]} error={error} onSubmit={onSubmitSpy} />);
    expect(wrapper).toMatchSnapshot();
});

test("Should trigger onNameChange", () => {
    const value = "NEW NAME";
    wrapper
        .find("input")
        .at(0)
        .simulate("change", {
            target: { value }
        });
    expect(wrapper.state("channel_name")).toEqual(value);
});

test("Should trigger onDeletedFlagChange", () => {
    wrapper
        .find("input")
        .at(1)
        .simulate("change", {});

    expect(wrapper.state("deleted_flag")).toEqual(true);
});

test("Should trigger onDescriptionChange", () => {
    const value = "NEW DESCRIPTION";
    wrapper.find("textarea").simulate("change", {
        target: { value }
    });
    expect(wrapper.state("channel_description")).toEqual(value);
});

test("Should trigger onsubmit", () => {
    localStorage.clear();
    localStorage.setItem("user_ID", 123);
    wrapper = shallow(<ChannelForm channel={channels[0]} error={error} onSubmit={onSubmitSpy} />);
    wrapper.find("form").simulate("submit", {
        preventDefault: () => {}
    });
    expect(wrapper.state("error")).toBe("");
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        channel_name: wrapper.state("channel_name"),
        deleted_flag: wrapper.state("deleted_flag"),
        channel_description: wrapper.state("channel_description"),
        user: localStorage.getItem("user_id")
    });
});
