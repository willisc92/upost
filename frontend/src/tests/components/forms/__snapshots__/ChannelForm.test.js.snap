// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ChannelForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  Channel Name:
   
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  Channel Description:
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <p>
    Visible 
    <span />
    <span>
      <input
        checked={true}
        id="string"
        name="prop1"
        onChange={[Function]}
        type="checkbox"
      />
    </span>
  </p>
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`Should render ChannelForm with existing channel data 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  Channel Name:
   
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value="Channel 1"
  />
  Channel Description:
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value="Test"
  />
  <p>
    Visible 
    <span />
    <span>
      <input
        checked={true}
        id="string"
        name="prop1"
        onChange={[Function]}
        type="checkbox"
      />
    </span>
  </p>
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;
